src/librustc/middle/infer/type_variable.rs

```diff
@@ -226,23 +224,26 @@ impl<'tcx> TypeVariableTable<'tcx> {
             .iter()
             .enumerate()
             .filter_map(|(i, value)| match &value.value {
-                &TypeVariableValue::Known(_) => None,
-                &TypeVariableValue::Bounded { .. } => Some(ty::TyVid { index: i as u32 })
-            })
-            .collect()
+                &TypeVariableValue::Bounded { .. } => Some(ty::TyVid { index: i as u32 }),
+                &TypeVariableValue::Known(v) => match v.sty {
+                    ty::TyInfer(ty::FloatVar(_)) | ty::TyInfer(ty::IntVar(_)) =>
```

Comment:
Nit: the behavior of this fn merits a comment. Also, `ty::TyInfer(_)` seems more "forward compatible" to me (or else adding the other cases with a call to bug, since I think they are impossible)